src/Workspaces/Core/Portable/Workspace/Solution/DocumentInfo.cs

```diff
@@ -234,7 +256,33 @@ Checksum IChecksummedObject.Checksum
                 {
                     if (_lazyChecksum == null)
                     {
-                        _lazyChecksum = Checksum.Create(nameof(DocumentAttributes), this);
+                        using (var stream = SerializableBytes.CreateWritableStream())
+                        using (var writer = new ObjectWriter(stream))
+                        {
+                            writer.WriteString(nameof(DocumentAttributes));
+
+                            if (FilePath == null || ProjectFilePath == null)
+                            {
+                                // this checksum is not persistable because
+                                // this info doesn't have non volatile info
+                                Id.WriteTo(writer);
```

Comment:
UGH.  i wish we had not allowed that.  

I still dont' like that we just silently switch to writing out the ID.  can we instead write out something that says "i'm anonymous, and i'm at this index in my parent?"